File: simple.js
const wordCounter = Object.create(null); //{} has protected and prepopulated keys, Map() is slower

require("readline").createInterface({input: process.stdin})
  .on("line", line => {
    line.toLowerCase().trim().split(' ').forEach(word => {
      wordCounter[word] = (wordCounter[word] || 0) + 1;
    });
  })
  .on("close", () => {
    let result = Object.entries(wordCounter)
      .sort((a, b) =>  b[1] - a[1])
      .map(entry => `${entry[0]} ${entry[1]}\n`)
      .join("");

    console.log(result);
  });

File: simple.dart
import 'dart:io';

void main() {
  final dict = <String, int>{};

  while (true) {
    String? input = stdin.readLineSync();
    if (input == null) break;
    input.split(" ").forEach((word) {
      if (word.isEmpty) return;
      final String lowerCaseWord = word.toLowerCase();
      dict[lowerCaseWord] = (dict[lowerCaseWord] ?? 0) + 1;
    });
  }

  var sortedKeys = dict.keys.toList()
    ..sort((a, b) => dict[b]!.compareTo(dict[a]!));
  sortedKeys.forEach((key) => print('$key ${dict[key]}'));
}
